api/spec/src/ratecards.tsp

```diff
@@ -74,69 +83,12 @@ model RateCardFlatFee extends RateCardBase {
    */
   @summary("Price")
   @example(#{ type: PriceType.flat, amount: "100", paymentTerm: "in_arrears" })
-  price: FlatPriceWithPaymentTerm | null;
-
-  /**
-   * The discount of the rate card. Either a percentage or an amount.
-   * Only available when price is set.
-   */
-  @summary("Discount")
-  discount?: DiscountPercentage | DiscountAmount;
-}
-
-/**
- * A usage-based rate card defines a price based on usage.
- */
-@friendlyName("RateCardUsageBased")
-model RateCardUsageBased extends RateCardBase {
-  type: RateCardType.usageBased;
-
-  /**
-   * The billing cadence of the rate card.
-   */
-  @summary("Billing cadence")
-  @encode(DurationKnownEncoding.ISO8601)
-  billingCandence: duration;
-
-  /**
-   * The price of the rate card.
-   * When null, the feature or service is free.
-   */
-  @summary("Price")
-  @example(#{ type: PriceType.unit, amount: "0.01", minimumAmount: "100" })
-  price: UnitPriceWithCommitments | TieredPriceWithCommitments | FlatPriceWithPaymentTerm | null;
-
-  /**
-   * The discount of the rate card.
-   */
-  @summary("Discount")
-  discount?: Discount;
-}
-
-/**
- * Rate card override.
- */
-@friendlyName("RateCardOverride")
-@discriminator("type")
-union RateCardOverride {
-  flatFee: RateCardOverrideFlatFee,
-  usageBased: RateCardOverrideUsageBased,
-}
-
-/**
- * Flat fee rate card override.
- */
-@friendlyName("RateCardOverrideFlatFee")
-model RateCardOverrideFlatFee {
-  ...PickProperties<RateCardFlatFee, "type" | "price" | "discount">;
-}
-
-/**
- * Usage-based rate card override.
- */
-@friendlyName("RateCardOverrideUsageBased")
-model RateCardOverrideUsageBased {
-  ...PickProperties<RateCardUsageBased, "type" | "price" | "discount">;
+  price:
+    | FlatPriceWithPaymentTerm
+    | UnitPriceWithCommitments
+    | TieredPriceWithCommitments
+    | FlatPriceWithPaymentTerm
+    | null;
```

Comment:
why not?

```
  price?:
    | FlatPriceWithPaymentTerm
    | UnitPriceWithCommitments
    | TieredPriceWithCommitments
    | FlatPriceWithPaymentTerm
```

Reply:
With `null`, you explicitly have to send `null` on API versus skipping the property.

Reply:
I believe it was the preference of @tothandras to use explicit nulls